wikimedia/wikipedia-ios
# [YiR] Present survey after last slide (Part 1)

**Phabricator:** https://phabricator.wikimedia.org/T376319

### Notes
This PR does most of the work for the survey view, however there will likely need to be some followup once the last donate slide is in.

Since this survey is presented after YiR is dismissed, it becomes another modal presentation on top of Article or Explore. I was a bit nervous about interceptions again between our fundraising campaigns (on Article) and all the other feature announcements, because the system will suppress one if more than one is attempted presented at the same time. So the last couple of commits are some cleanup I did to try funnel these presentations through one clear method (see [here](https://github.com/wikimedia/wikipedia-ios/blob/T376319/Wikipedia/Code/ExploreViewController.swift#L940) and [here](https://github.com/wikimedia/wikipedia-ios/blob/T376319/Wikipedia/Code/ArticleViewController.swift#L460)). LMK if there are questions or concerns!

### Test Steps
1. Fresh install app, ensure device region is a target country.
2. Build up 5+ articles on history tab.
3. Enable YiR developer settings flag in app settings.
4. Background / foreground to generate report.
5. Go to Explore tab. Note you may see some feature announcement modals here.
6. Go through YiR flow from Explore Profile.
7. On last slide, tap Done or Next.
8. Confirm YiR is dismissed and Survey is presented.
9. Ensure options are single-select. Tap Submit.
10. Ensure "Feedback submitted" toast appears.
11. Fresh install again, repeat steps 1-4. Go to Article Profile YiR. On last slide, tap Donate.
12. On last slide, tap Donate
13. Select a payment method option (if payment method sheet is presented).
14. Tap Back from donate form (to avoid making an actual donation)
15. Survey should appear after tapping Back.
16. Repeat Steps 7-11 for YiR from Article.
